src/transformStateWithClones.js

```diff
@@ -6,8 +6,38 @@
  *
  * @return {Object[]}
  */
+//
+//
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonedState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clonedState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete clonedState[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in clonedState) {
+          delete clonedState[key];
+        }
+        break;
+
+      default:
+        throw new Error('Unknown action type');
```

Comment:
Usually it's a bad idea to throw without any context, because this means that consumer of this function (You in the different part of the code, or someone else in the team) has to know that this function is not safe and has to check if it will not throw.
I would change it to `console.error('Your message')`
You can disable linting for this line only.

Reply:
Przepraszam, to była moja rekomendacja, a można więcej o tym, ze niebiezpieczne?